### PR TITLE
Add slot support for footer

### PR DESCRIPTION
**Pull request by samtsai**
While I understand your project has no use case for a summary row, I wanted to propose a very lightweight option to show `tfoot` data within a named slot.  When not using the slot, it will add harmless `<tfoot></tfoot>` markup as I thought a hide/show option was unnecessary.

Commit message:
Add test for `tfoot` slot
Minor lint cleanup

### CODE REVIEW

**Comment by sebastiandedeyne** [comment]
I don't mind this, what do you think @freekmurze ?

Maybe we should pass some data to the footer template though, so you can compute things in there like total count?

**Comment by freekmurze** [comment]
I'm ok with adding this. Passing all the data to the footer seems like a mighty fine idea.

**Comment by samtsai** [comment]
Let me add these things, update docs and tests.

File: src/components/TableComponent.vue [hunk 1]
```diff
@@ -38,6 +38,9 @@
                         :columns="columns"
                 ></table-row>
                 </tbody>
+                <tfoot>
+                    <slot name="tfoot" :rows="rows"></slot>
```
Comment (samtsai) [inline src/components/TableComponent.vue]: I'm passing `rows` as it seems to be the most useful information, but please suggest changes if there's a better property.

**Comment by samtsai** [comment]
I've updated demo, documentation, change log and tests.

**Comment by samtsai** [comment]
@sebastiandedeyne @freekmurze not exactly sure why tests are failing since I didn't change those areas. Looks like a similar issue with another PR (that's been closed #76 ) https://travis-ci.org/spatie/vue-table-component/builds/295341092

**Comment by sebastiandedeyne** [comment]
Gonna merge this and take care of Travis myself, thanks!

**Pull request merged**